### ISSUE TITLE
nil pointer dereference in http2 handler

### ISSUE THREAD

**Issue by doodles526** [superfly/wormhole#40]
New issue has arrisen. It appears that the fan-in listener might have a race where it can return a null `net.Conn` under certain circumstances. At least that's my current hypothesis from seeing

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x67a1cd]

goroutine 86 [running]:
net/http.(*conn).serve(0xc4202a01e0, 0xf4c880, 0xc4202cbaa0)
	/usr/lib/go/src/net/http/server.go:1691 +0x4d
created by net/http.(*Server).Serve
	/usr/lib/go/src/net/http/server.go:2720 +0x288
```

Where the given path errs at https://github.com/golang/go/blob/release-branch.go1.9/src/net/http/server.go#L2720

**Comment by doodles526** [comment]
Should note. This only occurs on client disconnection from heartbeat expiry

**Comment by doodles526** [comment]
Closed by #41
